plugins/matomo-backend/src/service/router.ts

```diff
@@ -0,0 +1,60 @@
+import { errorHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import express from 'express';
+import Router from 'express-promise-router';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+import { Logger } from 'winston';
+
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { config } = options;
+
+  const matomoToken = config.get('matomo.api_token') as string;
+  const matomoApiUrl = config.get('matomo.api_url') as string;
```

Comment:
Since you are pulling data from config, would you mind adding a `config.d.ts` to the backend plugin please? You can see OCM plugin for an example if you want:

https://github.com/janus-idp/backstage-plugins/blob/main/plugins/ocm-backend/config.d.ts

Please don't forget to add `config.d.ts` to the `package.json` as well.

Reply:
Yah sure. We didn't add config in backend as backend is default visible to whole config. 

But yah it will ensure validity

Reply:
Yeah, I guess, that is a fine approach when you use the plugin in a single instance 🙂 

Once published this is different, since user has to understand which config values are required and which are not. The schema ensures a valid config is passed to the instance when it is started.

Also, `ConfigReader` has a method `getString()`, so you don't need to cast the type (there's also `getOptionalString()` if the value is optional). See docs here:

https://backstage.io/docs/conf/reading#type-safety